src/mpi/init/init.c

```diff
@@ -153,28 +153,23 @@ int MPI_Init(int *argc, char ***argv)
     MPIR_ThreadInfo.isThreaded = 0;
 #endif /* MPICH_IS_THREADED */
 
-    mpi_errno = MPIR_T_env_init();
-    if (mpi_errno != MPI_SUCCESS)
-        goto fn_fail;
-
-    if (!strcasecmp(MPIR_CVAR_DEFAULT_THREAD_LEVEL, "MPI_THREAD_MULTIPLE"))
-        threadLevel = MPI_THREAD_MULTIPLE;
-    else if (!strcasecmp(MPIR_CVAR_DEFAULT_THREAD_LEVEL, "MPI_THREAD_SERIALIZED"))
-        threadLevel = MPI_THREAD_SERIALIZED;
-    else if (!strcasecmp(MPIR_CVAR_DEFAULT_THREAD_LEVEL, "MPI_THREAD_FUNNELED"))
-        threadLevel = MPI_THREAD_FUNNELED;
-    else if (!strcasecmp(MPIR_CVAR_DEFAULT_THREAD_LEVEL, "MPI_THREAD_SINGLE"))
-        threadLevel = MPI_THREAD_SINGLE;
-    else {
-        MPL_error_printf("Unrecognized thread level %s\n", MPIR_CVAR_DEFAULT_THREAD_LEVEL);
-        exit(1);
+    int threadLevel = MPI_THREAD_SINGLE;
+    const char *tmp_str;
+    if (MPL_env2str("MPIR_CVAR_DEFAULT_THREAD_LEVEL", &tmp_str)) {
```

Comment:
I see why this is done, but it makes me a little uneasy. We're losing some safety by managing this one cvar by hand vs. all the others. I guess I don't have a better solution at this point...

Reply:
Instead of view this as "managing" this one cvar, it is simply reading a environment variable -- independent of the cvar facility. The later `MPI_T_env_init` will manage all the cvars, including this one, -- which is also independent of whether the environment variable has been read before or not.